### ISSUE TITLE
check for type error 2322 does not catch uses of `any`.

### ISSUE THREAD

**Issue by snoble** [microsoft/TypeScript#49333]
# Bug Report

### 🔎 Search Terms

is:issue is:open "2322" "any" 

### 🕗 Version & Regression Information

Please keep and fill in the line that best applies:
-->
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about 4.6.4

### ⏯ Playground Link

[[Playground link with relevant code](https://www.typescriptlang.org/play?jsx=0&ssl=11&ssc=78&pln=9&pc=1#code/PQKgUALgngDgpgAgEIEMDOiC8CDeAjdOALgTQgCcBLAOwHMBfMAYwHtqyEANBbAHgBUEcAB4Q41ACZpkhAHwAKAG4l+ASh6yE8nFBLziMjKpUah9VWBDAwwYAn6xEAclQYnCStOosICdGkpaahQ8ABtECBYEaHgEJ34nADobOwQ41zh3Tz80AKCQ8OioiAALRFZ2ChQaXxYAM2jHOISAGgQ8AFdfePdWDtCJdsQaMhRqCEoUMUGAd0pSvwQJSjq6uHJxXzQOvBjEeoQKsnJq8fTCJPkAJgBmK6uLMGY2DgBNHgQBBWV7dUxNbS6LQGMZQYz2UxwczPSoIagkagdAC2eHWH1evERKPWCluqkSUHkACIyqFQiwiRYjixwolybR5NR1LZoiVstlvL4UHDkajyEJFOJWSwOrQSqyIk00FAyHAkaQUDKPL5PEA)]

### 💻 Code

```ts
const Y = <T>(v: T) => ({y: (e: any): T => e})
const n: number = Y<number>(23).y("hello")
console.log(n) // this is not a number even though the type system says it is
```

### 🙁 Actual behavior
The above passes the type checker even though it shouldn't. A similar version where `T extends Base` is caught. This is a bug since all types extend `any`.

### 🙂 Expected behavior

The type checker should catch the type error as it does below.

```ts
type Base = {base: string}
const X = <T extends Base>(v: T) => ({y: (e: Base): T => e})
```
This is caught by the type checker with error
```
Type 'Base' is not assignable to type 'T'.
  'Base' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint 'Base'.(2322)
```

**Comment by snoble** [comment]
not actually a bug
